pkg/deployer/helm/ensure.go

```diff
@@ -119,10 +119,35 @@ func (h *Helm) ApplyFiles(ctx context.Context, files map[string]string, exports
 func (h *Helm) CheckResourcesHealth(ctx context.Context, client client.Client) error {
 
 	if !h.ProviderConfiguration.HealthChecks.DisableDefault {
-		err := h.defaultCheckResourcesHealth(ctx, client)
+		defaultHealthCheck := healthcheck.DefaultHealthCheck{
+			Context:          ctx,
+			Client:           client,
+			CurrentOp:        "DefaultCheckResourcesHealthHelm",
+			Log:              h.log,
+			Timeout:          h.ProviderConfiguration.HealthChecks.Timeout,
+			ManagedResources: h.ProviderStatus.ManagedResources,
+		}
+		err := defaultHealthCheck.CheckResourcesHealth()
 		if err != nil {
 			return err
 		}
+	} else if h.ProviderConfiguration.HealthChecks.CustomHealthChecks != nil {
+		//TODO: thread this
```

Comment:
with thread you mean parallelize?

Reply:
Yeah, I meant to put the custom health checks into go() functions so that they can run in parallel instead of sequentially. However, the custom health check calls `WaitForObjectsHealthy()` which already is parallelized (it calls `IsObjectHealthy() inside a go func(){} block) and therefore, I believe it is safe to just remove this comment. Therefore, I do not see any benefit in kicking off the custom healthchecks in parallel and and I will just remove the comment.

Reply:
Comment removed.